frontends/web/src/components/TaskLeaderboard/TaskModelLeaderboardTable.js

```diff
@@ -168,9 +168,11 @@ const TaskModelLeaderboardRow = ({ model, metrics, datasetWeights }) => {
               {model.model_name}
             </Link>
           ) : (
-            "Anonymous Model " + model.model_id
+            "Unpublished Model " + model.model_id
           )}{" "}
-          {model.username && (
+          {model.username && model.username === "anonymous" ? (
```

Comment:
Can we make this "if" block about model.uid instead of model.username? If we make it about username, then it won't link to somebody with a username "anonymous"